lib/job.js

```diff
@@ -1,106 +1,249 @@
 'use strict';
 
+const bluebird = require('bluebird');
+const Path = require('path');
+const logger = require('./logger.js');
+
+function Job(client, config) {
```

Comment:
Yup. I'll be `class`-ing all the things. For whatever reason, I thought `class` was only available in >6, but I guess I was wrong.